app/models/account/provider_methods.rb

```diff
@@ -67,6 +67,7 @@ def build_kind(kind:, **attributes)
       end
     end
 
+    # has_many :buyer_accounts, -> { not_master }, class_name: 'Account', foreign_key: 'provider_account_id', dependent: :destroy, inverse_of: :provider_account do
     has_many :buyer_accounts, ->(provider) { where.not(id: provider.id) }, class_name: 'Account', foreign_key: 'provider_account_id', dependent: :destroy, inverse_of: :provider_account do
```

Comment:
This is more like a question - which one is better. Initially I changed with the commented out version because joins are not possible with associations taking a parameter. But then I worked-around by conditionally ecluding them. 

So this change is not needed but I remained wondering whether it's better to change or not.

The point of the original version is to prevent master seeing itself as a buyer account. Because master's `provider_account_id` is same as its `id`. All other accounts should have a different `id` than their `provider_account_id`.

That's why in the commented out version it is enough to filter out the master account. For now I'm inclined to leave it as is.

Reply:
I don't have a strong opinion, but if both are equally valid (in results, performance, etc), then I'd took the commented one because it's easier to understand and use.

Reply:
I'll leave it for another PR. Thank you.